starfish/test/image/test_physical_coordinate_calculator.py

```diff
@@ -0,0 +1,184 @@
+                    PhysicalCoordinateTypes.X_MIN.value,
+                    PhysicalCoordinateTypes.X_MAX.value,
+                    PhysicalCoordinateTypes.Y_MIN.value,
+                    PhysicalCoordinateTypes.Y_MAX.value,
+                    PhysicalCoordinateTypes.Z_MIN.value,
+                    PhysicalCoordinateTypes.Z_MAX.value,
+                ],
+            },
+        )
+
+        self.indexers = {Indices.ROUND: slice(None, None),
+                         Indices.CH: slice(None, None),
+                         Indices.Z: slice(None, None),
+                         Indices.Y: slice(None, None),
+                         Indices.X: slice(None, None)}
+
+        self.physical_coords = {PhysicalCoordinateTypes.X_MIN: 1,
+                                PhysicalCoordinateTypes.X_MAX: 2,
+                                PhysicalCoordinateTypes.Y_MIN: 4,
+                                PhysicalCoordinateTypes.Y_MAX: 6,
+                                PhysicalCoordinateTypes.Z_MIN: 1,
+                                PhysicalCoordinateTypes.Z_MAX: 3}
+
+        self.coords_array.loc[0, 0, 0] = \
+            np.array([self.physical_coords[PhysicalCoordinateTypes.X_MIN],
+                      self.physical_coords[PhysicalCoordinateTypes.X_MAX],
+                      self.physical_coords[PhysicalCoordinateTypes.Y_MIN],
+                      self.physical_coords[PhysicalCoordinateTypes.Y_MAX],
+                      self.physical_coords[PhysicalCoordinateTypes.Z_MIN],
+                      self.physical_coords[PhysicalCoordinateTypes.Z_MAX]])
+
+        self.stack_shape = OrderedDict([(Indices.ROUND, 1), (Indices.CH, 1),
+                                        (Indices.Z, 1), (Indices.Y, 200), (Indices.X, 200)])
+
+    def test_calc_new_physical_coords_array(self):
+        # Calculate physical pixel sizes for x and y
+        physical_pixel_size_x = \
+            physical_coordinate_calculator.\
+            calculate_physcial_pixel_size(
+                coord_max=self.physical_coords[PhysicalCoordinateTypes.X_MAX],
+                coord_min=self.physical_coords[PhysicalCoordinateTypes.X_MIN],
+                num_pixels=self.stack_shape[Indices.X])
+
+        physical_pixel_size_y = \
+            physical_coordinate_calculator.\
+            calculate_physcial_pixel_size(
+                coord_max=self.physical_coords[PhysicalCoordinateTypes.Y_MAX],
+                coord_min=self.physical_coords[PhysicalCoordinateTypes.Y_MIN],
+                num_pixels=self.stack_shape[Indices.Y])
+
+        # Index on single value of X, range of y
+        self.indexers[Indices.Y], self.indexers[Indices.X] = 100, slice(None, 100)
+        new_coords = physical_coordinate_calculator.calc_new_physical_coords_array(
+            physical_coordinates=self.coords_array,
+            stack_shape=self.stack_shape,
+            indexers=self.indexers)
+
+        expected_xmin = physical_coordinate_calculator.calculate_physical_pixel_value(
```

Comment:
I wonder if it makes sense to calculate these by hand and store the values in the test file.

Reply:
haaaaa I had a version of this file that did that but thought you might tell me to just use the method. Predicted wrong :(

Reply:
I can see advantages to both, which is why I'm just wondering! :)